File: Carly_Michael/jectacular-ql/src/app/domain/ast/qls/qls-question.ts
import {Location} from '../location';
import {Widget} from './widget';
import {QlsNode} from './qls-node';
import {QlsVisitor} from './visitors/qls-visitor';

export class QlsQuestion extends QlsNode {
  constructor(readonly name: string, public widget: Widget, readonly location: Location) {
    super();
  }

  accept<T>(visitor: QlsVisitor<T>): T {
    return visitor.visitQlsQuestion(this);
  }
}

File: GunOli/src/main/java/QLS/ParseObjectQLS/QLSQuestion.java
package QLS.ParseObjectQLS;

import QLS.Analysis.WidgetVisitorInterface;
import QLS.ParseObjectQLS.Widgets.Widget;

import java.util.ArrayList;

public class QLSQuestion extends QLSNode {

    private String identifier;
    private ArrayList<Widget> widgets;


    public QLSQuestion(String identifier, ArrayList<Widget> widgets, int line){
        super(line);
        this.identifier = identifier;
        this.widgets = widgets;
    }

    public String getIdentifier() {
        return identifier;
    }

    public ArrayList<Widget> getWidget() {
        return widgets;
    }

    @Override
    public <T> T accept(WidgetVisitorInterface<T> visitor) {
        return visitor.visit(this);
    }
}
